# Analysis

Create your own EDA below! Create as many code-blocks as you need.

Once you've completed your EDA, complete the section titled **Reflections** where you will answer questions about your EDA.

In [1]:
import pandas as pd 
import numpy as np

import seaborn as sns
import matplotlib.pyplot as plt

In [4]:
# import coffee data
coffee = pd.read_csv("../data/crop/coffee_output.csv")

# import weather data
weather = pd.read_csv("../data/weather/weather_data.csv")

In [9]:
# TODO: Merge these two dataframes together

merged_df = pd.merge(weather, coffee, on=["year", "subdivision"])


In [10]:
# TODO: View the first 5 rows of your data

merged_df.head()

year  rain_max   temp_avg   temp_max   temp_min    hum_max    hum_min  \
0  2006  6.670968  21.803087  27.190323  18.654839  97.129032  57.806452   
1  2007  1.866250  23.040999  27.649375  18.753125  87.837500  48.493750   
2  2008  3.777049  20.682400  27.000000  16.216393  96.136612  51.415301   
3  2009  3.380220  21.023802  27.769231  16.410440  96.054945  49.329670   
4  2010  2.183516  21.444293  28.580769  16.156044  95.247253  44.648352   

   wind_max  wind_avg   subdivision country     type  60kgs_bag  \
0  9.706452  2.075207  Minas Gerais  Brazil  Arabica      16.30   
1  8.749375  2.069605  Minas Gerais  Brazil  Arabica      23.70   
2  8.865027  1.584408  Minas Gerais  Brazil  Arabica      16.85   
3  9.114286  1.611746  Minas Gerais  Brazil  Arabica      25.55   
4  9.109341  1.455907  Minas Gerais  Brazil  Arabica      22.10   

   nonbearing_trees  bearing_trees  nonbear_hectares  bearing_hectares  \
0             581.0         5330.0             175.0            2298.0   
1             563.0         5730.0             186.0            2245.0   
2             679.0         5720.0             196.0            2235.0   
3             663.0         5770.0             188.0            2223.0   
4             873.0         5725.0             244.0            2151.0   

   bearing_trees_per_hectare  nonbearing_trees_per_hectare  
0                     2319.0                        3320.0  
1                     2552.0                        3030.0  
2                     2559.0                        3468.0  
3                     2596.0                        3527.0  
4                     2662.0                        3578.0

In [11]:
# TODO: Check for null values 

merged_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 18 entries, 0 to 17
Data columns (total 19 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   year                          18 non-null     int64  
 1   rain_max                      18 non-null     float64
 2   temp_avg                      18 non-null     float64
 3   temp_max                      18 non-null     float64
 4   temp_min                      18 non-null     float64
 5   hum_max                       18 non-null     float64
 6   hum_min                       18 non-null     float64
 7   wind_max                      18 non-null     float64
 8   wind_avg                      18 non-null     float64
 9   subdivision                   18 non-null     object 
 10  country                       18 non-null     object 
 11  type                          18 non-null     object 
 12  60kgs_bag                     18 non-null     float64
 13  nonbear

In [12]:
# TODO: Drop rows that contain null values

merged_df.dropna(inplace=True)
merged_df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 17 entries, 0 to 17
Data columns (total 19 columns):
 #   Column                        Non-Null Count  Dtype  
---  ------                        --------------  -----  
 0   year                          17 non-null     int64  
 1   rain_max                      17 non-null     float64
 2   temp_avg                      17 non-null     float64
 3   temp_max                      17 non-null     float64
 4   temp_min                      17 non-null     float64
 5   hum_max                       17 non-null     float64
 6   hum_min                       17 non-null     float64
 7   wind_max                      17 non-null     float64
 8   wind_avg                      17 non-null     float64
 9   subdivision                   17 non-null     object 
 10  country                       17 non-null     object 
 11  type                          17 non-null     object 
 12  60kgs_bag                     17 non-null     float64
 13  nonbearing_t

In [ ]:
# TODO: BEGIN YOUR EDA!



## Reflections

Answer each question based on the visualizations that you've generated.

Remember, you must justify your answers with proof! Look back to W6D3 for more context.

In [13]:
temp_bags = merged_df["temp_avg"].corr(merged_df["60kgs_bag"])
temp_bags

np.float64(-0.12477217306378187)

What is the relationship between temp_avg and 60kgs_bag? How *strong* is this relationship? Use pearson's correlation coefficient.

> The correlation between both is -0.12. This is weakly negative.

In [14]:
hum_bags = merged_df["hum_min"].corr(merged_df["60kgs_bag"])
hum_bags

np.float64(-0.5401709088326815)

What is the relationship between minimum humidity and 60kgs_bag? How *strong* is this relationship? Use pearson's correlation coefficient.

> The correlation between both is -0.54. This is moderately negative.

In [15]:
rain_bags = merged_df["rain_max"].corr(merged_df["60kgs_bag"])
rain_bags

np.float64(-0.3958109913734919)

What is the relationship between rain_max and 60kgs_bag? How *strong* is this relationship? Use pearson's correlation coefficient.

> The correlation between both is -0.39. This is weakly negative.

In [16]:
wind_bags = merged_df["wind_avg"].corr(merged_df["60kgs_bag"])
wind_bags

np.float64(-0.7073271023009807)

What is the relationship between average wind speed and 60kgs_bag? How *strong* is this relationship? Use pearson's correlation coefficient.

> The correlation between both is -0.70. This is strongly negative.

In [17]:
col_1 = ["temp_avg", "hum_min", "rain_max", "wind_avg"]
pearson_corr = []
for col in col_1:
    pearson_corr.append(merged_df[col].corr(merged_df["bearing_trees_per_hectare"]))
print(pearson_corr)

[np.float64(-0.43207570398850825), np.float64(-0.5512035768352691), np.float64(-0.25352398713240387), np.float64(-0.8939870098310204)]


Perform the same bivariate analysis between the mentioned weather columns (temp_avg, hum_min, rain_max, and wind_avg) and a coffee-related column of your choice (bearing_trees, nonbearing_trees, bearing_trees_per_hectare, etc). Which directions do these relationships follow? What are the strengths of these relationships?

> All relationships were negative against brearing trees per hectare. These are the results: -0.43 (moderately negative vs temp_avg); -0.55 (moderately negative vs hum_min); -0.25 (weakly negative vs rain_max); -0.89 (strongly negative vs wind_avg)

Which weather factor seems to influence coffee production the most? 

> Only wind had both strong correlation coefficients of -0.70 vs 60kgs_bag and -0.89 vs bearing_trees_per_hectare. Both correlations are negative. High wind is the strongest (negative) factor to coffee production.

Which data would you collect to continue exploring the impacts of weather and economic forces on coffee production?

Answer here

If you were preparing a report to provide coffee farmers, which weather condititions would you state are the most favorable for coffee growing (assuming any exist)?

Answer here